internal/pkg/cli/task_run.go

```diff
@@ -538,10 +546,10 @@ func (o *runTaskOpts) runTaskCommand() (cliStringer, error) {
 			return nil, err
 		}
 	case 3:
-		appName, envName, serviceName := parts[0], parts[1], parts[2]
-		cmd, err = o.runTaskRequestFromService(ecs.New(sess), appName, envName, serviceName)
+		appName, envName, workloadName := parts[0], parts[1], parts[2]
+		cmd, err = o.runTaskCommandFromWorkload(sess, appName, envName, workloadName)
 		if err != nil {
-			return nil, fmt.Errorf("generate task run command from service %s of application %s deployed in environment %s: %w", serviceName, appName, envName, err)
+			return nil, err
 		}
 	default:
 		return nil, errors.New("invalid input to --generate-cmd: must be of one the form <cluster>/<service> or <app>/<env>/<workload>")
```

Comment:
nit (sorry this is in existing code; I didn't see it earlier):
```suggestion
		return nil, errors.New("invalid input to --generate-cmd: must be of format <cluster>/<service> or <app>/<env>/<workload>")
```
(Here and in the unit test.)